packages/ui-components/src/components/toggle-tip/toggle-tip.tsx

```diff
@@ -0,0 +1,104 @@
+import { autoPlacement, computePosition, ComputePositionConfig, offset, shift } from '@floating-ui/dom';
+import { Component, Host, h, Prop, Element, Listen, Event, EventEmitter } from '@stencil/core';
+import { isEmpty } from 'lodash-es';
+import { ETooltipPosition } from '../../utils/types/components';
+import { DEFAULT_AUTOPLACEMENT_CONFIG, DEFAULT_OFFSET, DEFAULT_SHIFT_CONFIG, DEFAULT_TOOLTIP_CONFIG } from '../tooltip/tooltip.config';
+import { IToggleTip, IToggleTipEvents } from './toggle-tip.types';
+
+/**
+ * @part toggle-tip-container - The toggle tip container.
+ * @part toggle-tip-slot-content - The toggle tip slot content.
+ */
+@Component({
+	tag: 'kv-toggle-tip',
+	styleUrl: 'toggle-tip.scss',
+	shadow: true
+})
+export class KvToggleTip implements IToggleTip, IToggleTipEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) text?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) position?: ETooltipPosition;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) allowedPositions?: ETooltipPosition[];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) options?: Partial<ComputePositionConfig>;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) isOpen?: boolean = false;
+
+	/** @inheritdoc */
+	@Event() openStateChange: EventEmitter<boolean>;
+
+	/** The Host's element reference */
+	@Element() el: HTMLKvToggleTipElement;
+
+	@Listen('click', { target: 'window' })
+	checkForClickOutside(ev: { target: Node }) {
+		if (this.el.contains(ev.target)) {
+			return;
+		}
+		const toggletip = this.el.shadowRoot.querySelector('#toggle-tip') as HTMLElement;
+		toggletip.style.display = '';
+		this.isOpen = !this.isOpen;
+		this.openStateChange.emit(this.isOpen);
+		this.calculateTogglePositions();
+	}
+
+	componentDidRender() {
+		this.calculateTogglePositions();
+	}
+
+	private calculateTogglePositions = () => {
+		const toggleIcon = this.el.shadowRoot.querySelector('#toggle-tip-content');
+		const toggletip = this.el.shadowRoot.querySelector('#toggle-tip') as HTMLElement;
+
+		const position = isEmpty(this.allowedPositions) ? this.position : undefined;
+		const middleware = [offset(DEFAULT_OFFSET), shift(DEFAULT_SHIFT_CONFIG)];
+
+		if (isEmpty(position)) {
+			middleware.push(
+				autoPlacement({
+					...DEFAULT_AUTOPLACEMENT_CONFIG,
+					allowedPlacements: this.allowedPositions
+				})
+			);
+		}
+
+		const onButtonClick = () => {
```

Comment:
I was wondering if it's possible to move this function to out of this scope, but seems difficult. Only if we bind the callback with arguments that we need to run the floating/ui